examples/quickstart/main.rs

```diff
@@ -33,16 +62,37 @@ async fn main() {
     let tycho_api_key: String =
         env::var("TYCHO_API_KEY").unwrap_or_else(|_| "sampletoken".to_string());
 
-    let tvl_threshold = 10_000.0;
-    let tvl_filter = ComponentFilter::with_tvl_range(tvl_threshold, tvl_threshold);
+    let cli = Cli::parse();
+    let tvl_filter = ComponentFilter::with_tvl_range(cli.tvl_threshold, cli.tvl_threshold);
 
     let all_tokens =
         load_all_tokens(tycho_url.as_str(), false, Some(tycho_api_key.as_str()), Chain::Ethereum)
             .await;
-    let mut pairs: HashMap<String, Vec<Token>> = HashMap::new();
+
+    let sell_token_address =
+        Bytes::from_str(&cli.sell_token).expect("Invalid address for sell token");
+    let buy_token_address = Bytes::from_str(&cli.buy_token).expect("Invalid address for buy token");
+    let sell_token = all_tokens
+        .get(&sell_token_address)
+        .expect("Sell token not found")
+        .clone();
+    let buy_token = all_tokens
+        .get(&buy_token_address)
+        .expect("Buy token not found")
+        .clone();
+    let amount_in =
+        BigUint::from(cli.sell_amount) * BigUint::from(10u32).pow(sell_token.decimals as u32);
+
+    println!(
+        "Looking for the best swap for {} {} -> {}",
+        cli.sell_amount, sell_token.symbol, buy_token.symbol
+    );
+    let mut pairs: HashMap<String, ProtocolComponent> = HashMap::new();
+    let mut amounts_out: HashMap<String, BigUint> = HashMap::new();
 
     let mut protocol_stream = ProtocolStreamBuilder::new(&tycho_url, Chain::Ethereum)
         .exchange::<UniswapV2State>("uniswap_v2", tvl_filter.clone(), None)
+        .exchange::<UniswapV3State>("uniswap_v3", tvl_filter.clone(), None)
```

Comment:
@tamaralipows this is not working for uniswap v3 yet. There is a problem when decoding the fee 😕 I thought it was only the key that was wrong (and changed it [here](https://github.com/propeller-heads/tycho-execution/pull/57/commits/5eaf5315716dc481a00248c3b21f4a75a9c23b6b)) but it seems we are decoding wrong too. If you have some time, can you have a look please? Just run the quickstart in this branch. Right before it fails to encode you can print the whole component to see how the fee really looks like!

Reply:
Just saw this now - I will start tomorrow by looking into it then!